### PR TITLE
emit nice summary at the end of the release (release notes)

### PR DESCRIPTION
**Pull request by epeee**
- very first version which includes a link to the release notes

first part of #450 

Did some testing locally:

```
:performRelease

Release shipped!
    - Release notes:      https://github.com/mockito/shipkit/blob/ep2/docs/release-notes.md
```

### CODE REVIEW

File: subprojects/shipkit/src/main/groovy/org/shipkit/internal/notes/util/Supplier.java [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package org.shipkit.internal.notes.util;
+
+public interface Supplier<T> {
```
Comment (szczepiq) [inline subprojects/shipkit/src/main/groovy/org/shipkit/internal/notes/util/Supplier.java]: Can you add a TODO to remove when we migrate to Java8?

I like this new interface. I would introduce it myself :)

File: subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/util/TaskSuccessfulMessage.java [hunk 2]
```diff
@@ -21,4 +22,13 @@ public void execute(Task task) {
             }
         });
     }
+
+    public static void logOnSuccess(Task task, final Supplier<String> supplier) {
```
Comment (szczepiq) [inline subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/util/TaskSuccessfulMessage.java]: Can you add short Javadoc why we need this method (and also in the other method on this class)?

Good stuff. I'm happy that you have figured out how to defer logging the message, until we have all information needed.

**Comment by szczepiq** [comment]
@epeee, do you mind addressing PR comments from my review?

Merged with pleasure!

**Comment by epeee** [comment]
@szczepiq thx for your feedback!
I will apply your suggestions soon.

**Pull request merged**